ReactAndroid/src/main/java/com/facebook/react/modules/websocket/WebSocketModule.java

```diff
@@ -65,12 +68,46 @@ public void connect(final String url, final int id) {
     // Disable timeouts for read
     client.setReadTimeout(0, TimeUnit.MINUTES);
 
-    Request request = new Request.Builder()
+    Request.Builder builder = new Request.Builder()
         .tag(id)
-        .url(url)
-        .build();
+        .url(url);
 
-    WebSocketCall.create(client, request).enqueue(new WebSocketListener() {
+    if (options != null) {
+      if (options.hasKey("origin")) {
+        builder.addHeader("Origin", options.getString("origin"));
+      }
+
+      if (options.hasKey("headers")) {
+        ReadableMap headers = options.getMap("headers");
+        ReadableMapKeySetIterator keyIterator = headers.keySetIterator();
+        while (keyIterator.hasNextKey()) {
+          String key = keyIterator.nextKey();
+          switch (headers.getType(key)) {
+            case Null:
+              builder.addHeader(key, "null");
+              break;
+            case Boolean:
+              builder.addHeader(key, String.valueOf(headers.getBoolean(key)));
+              break;
+            case Number:
+              // Can be int or double.
+              builder.addHeader(key, String.valueOf(headers.getDouble(key)));
+              break;
+            case String:
+              builder.addHeader(key, headers.getString(key));
+              break;
+            case Map:
+              throw new UnsupportedOperationException("Maps aren't supported header values.");
+            case Array:
+              throw new UnsupportedOperationException("Arrays aren't supported header values.");
```

Comment:
Can you add a TODO comment to support this? For other APIs, arrays of header values are joined by `", "`.